www/js/app/common/service/common-service.js

```diff
@@ -0,0 +1,44 @@
+var movieServices = angular.module('movieServices');
+movieServices.service('query', ['$resource', 'APP_CONSTANTS',  function($resource, APP_CONSTANTS) {
+	return {
+			'movieUpcoming' : $resource(APP_CONSTANTS.WS_URL +
+										APP_CONSTANTS.MOVIE_UPCOMING +
+										APP_CONSTANTS.API_KEY +
+										APP_CONSTANTS.PARAM_PAGE+':page'),
+
+	 		'genreMovieList': $resource(APP_CONSTANTS.WS_URL +
```

Comment:
What would happen if the endpoints of our services change while using $http? We would have to change in several places of the application, with $resource we'd change only in one spot. Another point is that some endpoints were born of the concatenation of strings, something very likely to become a problem. These issues appear because $http is not able to hide part of the complexity that is dealing with our REST endpoints.
One of the great advantages of using $resource is that it provides a high level of abstraction over its similar $http, making it easier when you're working with a RESTful back-end.